### ISSUE TITLE
Photo Picker

### ISSUE THREAD

**Issue by zhouzhuli** [yasirkula/UnityNativeGallery#332]
In order to protect user information, Google will no longer support directly accessing the contents of the album for non-image applications. Instead, it will use the photo picker to retrieve specified photos. Have you considered making corresponding updates? For example, can you choose between the original method or the new method?

**Comment by yasirkula** [comment]
NativeGallery uses an Intent to request the image from the gallery app. It doesn't create its own gallery user interface so it doesn't  need to have direct access to images. If this method fails in the future, then I shall revise it.

**Comment by book233** [comment]
Google's policy states September 18, 2024 - October 31, 2024: READ_MEDIA_IMAGES Developers of apps with permissions and READ_MEDIA_VIDEOpermissions will be prompted in the Google Play Console to either submit a declaration form to be approved for core/broad access use cases, or to remove the permission (for one-time/infrequent use cases). Different actions are required for one-time/infrequent use cases and for use in core/broad access use cases. This is a bit of a headache for us who don't use the system album frequently. Is it possible for you to make corresponding optimizations for this?

**Comment by yasirkula** [comment]
@book233 I don't use these permissions.

**Comment by book233** [comment]
![企业微信截图_a19ce33c-dfa9-42e2-8630-60d74117c5cd](https://github.com/user-attachments/assets/6b70464b-d6ba-42f4-aaba-6c0e6be944fd)
I used NativeGallery 1.7.1 to check the AndroidManifest in the aar and found this. It seems that there are related permissions declared?

**Comment by yasirkula** [comment]
Yeah they used to exist but I've removed them in v1.7.6. Updating the plugin should resolve the problem.

**Comment by ArkTarusov** [comment]
All my apps are getting rejected by Google because this plugin doesn't work correctly with galleries.

Google has been requiring Photo Picker since ~2023, starting with Android 13.
This month, this requirement became mandatory.

**Comment by yasirkula** [comment]
@ArkTarusov Can you copy the full rejection message?

**Comment by ArkTarusov** [comment]
@yasirkula 

> Photo and Video Permissions policy: Permission use is not directly related to your app’s core purpose.
> We found that your app is not compliant with how the READ_MEDIA_IMAGES/READ_MEDIA_VIDEO permissions are allowed to be used.
> 
> Your app only requires one-time or infrequent access to media files on the device. Only apps with a core use case that require persistent access to photo and video files located in shared storage on devices are allowed to use photo and video permissions. For more details on the requirements, please see Google Play's Photo and Video Permissions policy.
> 
> To comply with Google Play's Photo and Video Permissions policy, please adjust the following requirements.
> If your app requires one-time, or limited use of photo and video file, remove the permissions and consider using the Android photo picker.

Upon contacting Google, it was confirmed that file manager applications and similar software are the only entities permitted to utilize these permissions. Games and all other applications are now restricted from accessing files; access is exclusively permitted through the Photo Picker, while all other methods are prohibited.

**Comment by yasirkula** [comment]
@ArkTarusov If your plugin is up-to-date, these permissions can't come from NativeGallery since they've been removed ~2.5 years ago. I'm using ACTION_PICK and ACTION_GET_CONTENT to pick images and as long as these aren't deprecated, you should be fine.

**Comment by ArkTarusov** [comment]
@yasirkula That's not entirely accurate. This plugin indirectly requests restricted permissions through a chain of interactions.

Specifically, when a user tries to select a photo via the plugin, they are prompted to choose an external app. That app, in turn, requires access to media files. Google now interprets this flow as the main app effectively accessing restricted media permissions.

I implemented Photo Picker, and now the user is not prompted for any permissions at all. We also successfully passed additional Google review checks and were republished.

For reference:
- All plugins in our project were updated on time.
- Google had been notifying us for several months about the need to replace the photo access approach.
- We received the final warning while using version 1.9.1. Updating the plugin to version 1.9.2 did not resolve the issue or help us pass Google's review.
- We did not remove the plugin entirely. We still use the SaveToGallery functionality on Android and retain full functionality on iOS.
- A custom implementation was introduced specifically for Photo Picker and is now used instead of GetImageFromGallery.

**Comment by yasirkula** [comment]
@ArkTarusov I have no experience with using AndroidX (support library). How did you implement the Photo Picker and use AndroidX in Unity? Was the process hard? If I did the same thing, would installing NativeGallery still be user-friendly?

**Comment by ArkTarusov** [comment]
@yasirkula Overall, I didn’t encounter any issues during implementation, although I’m not a professional Android developer.

I needed to urgently comply with a Google requirement, so my current implementation differs from the NativeGallery API. At this stage, it has only been tested for image selection.

I used Kotlin and integrated the native code into Unity via an .androidlib module. The library includes an Activity that extends AppCompatActivity and serves as an entry point for invoking PickVisualMediaRequest.

I plan to submit a PR within the next two or three days to add support for selecting images via Photo Picker.
I will also document the known issues. 
I’d appreciate any feedback on this functionality, as this is not an area I work in frequently.
